components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2827,6 +2831,38 @@ public API createNewAPIVersion(String existingApiId, String newVersion, Boolean
         return getAPIbyUUID(newAPIId, organization);
     }
 
+    private void attachOperationPoliciesToAPI(API newAPI, Map<String, String> clonedOperationPolicyMap,
+                                              Map<String, List<OperationPolicy>> operationPoliciesMap)
+            throws APIManagementException {
+        operationPoliciesMap.forEach((key, operationPolicies) ->
+                operationPolicies.forEach(operationPolicy ->
+                        operationPolicy.setPolicyId(clonedOperationPolicyMap.get(operationPolicy.getPolicyId()))));
+        Set<URITemplate> uriTemplates = newAPI.getUriTemplates();
+        if (uriTemplates != null) {
+            for (URITemplate uriTemplate : uriTemplates) {
+                List<OperationPolicy> operationPolicies =
+                        operationPoliciesMap.get(uriTemplate.getHTTPVerb() + ":" + uriTemplate.getUriTemplate());
+                uriTemplate.setOperationPolicies(operationPolicies);
+            }
+            apiMgtDAO.addOperationPolicyMapping(uriTemplates);
+        }
+    }
+
+    private Map<String, String> cloneOperationPoliciesToAPI(API newAPI, Map<String, List<OperationPolicy>> operationPoliciesMap)
+            throws APIManagementException {
+        Map<String, String> clonedPolicies = new HashMap<>();
+        for (Map.Entry<String, List<OperationPolicy>> operationPolicyEntry : operationPoliciesMap.entrySet()) {
+            List<OperationPolicy> operationPolicyList = operationPolicyEntry.getValue();
+            for (OperationPolicy operationPolicy : operationPolicyList) {
+                if (clonedPolicies.containsKey(operationPolicy.getPolicyId())) {
+                    String policyUUID = apiMgtDAO.cloneOperationPolicy(operationPolicy.getPolicyId(), newAPI.getUuid(), null);
```

Comment:
Shall we get the previous APIs uuid to the method and use it to get APISpecificpolicy and add it as a new API specific policy under the new API? This way the policy artifact will include the cloned common policy Id.